pilot/pkg/xds/sds.go

```diff
@@ -60,10 +60,7 @@ func (sr SecretResource) Cacheable() bool {
 	return true
 }
 
-func sdsNeedsPush(proxy *model.Proxy, updates model.XdsUpdates) bool {
-	if proxy.Type != model.Router {
-		return false
-	}
+func sdsNeedsPush(updates model.XdsUpdates) bool {
 	if len(updates) == 0 {
 		return true
```

Comment:
No update here so need push?

Reply:
Existing code, have not modified anything, except removing the proxy type check

Reply:
len(updates) == 0 does not mean no config update, could be a global config like meshconfig/network update.